autonomy/chain/service.py

```diff
@@ -40,16 +42,50 @@
 
 DEFAULT_DEPLOY_PAYLOAD = "0x"
 
+ServiceInfo = Tuple[int, str, bytes, int, int, int, int, List[int]]
+
+
+def get_service_info(
+    ledger_api: LedgerApi, chain_type: ChainType, token_id: int
+) -> ServiceInfo:
+    """
+    Returns service info.
+
+    :param ledger_api: `aea.crypto.LedgerApi` object for interacting with the chain
+    :param chain_type: Chain type
+    :param token_id: Token ID pointing to the on-chain service
+    :returns: security deposit, multisig address, IPFS hash for config,
+            threshold, max number of agent instances, number of agent instances,
+            service state, list of cannonical agents
+    """
+
+    return registry_contracts.service_registry.get_service_information(
+        ledger_api=ledger_api,
+        contract_address=ContractConfigs.get(SERVICE_REGISTRY_CONTRACT.name).contracts[
+            chain_type
+        ],
+        token_id=token_id,
+    )
+
 
 def activate_service(
     ledger_api: LedgerApi,
     crypto: Crypto,
     chain_type: ChainType,
     service_id: int,
-    bond_value: int,
 ) -> None:
     """Activate service."""
 
+    (cost_of_bond, *_, service_state, _,) = get_service_info(
+        ledger_api=ledger_api, chain_type=chain_type, token_id=service_id
```

Comment:
Instead of getting cost of bond as input from the user we retrieve it from on-chain to activate, register and deploy service